src/benchmarks/micro/libraries/System.Numerics.BitOperations/Perf_BitOperations.cs

```diff
@@ -0,0 +1,96 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Numerics;
+using BenchmarkDotNet.Attributes;
+using MicroBenchmarks;
+using System.Runtime.CompilerServices;
+
+namespace System.Numerics.Tests
+{
+    [BenchmarkCategory(Categories.Libraries, Categories.SIMD)]
+    public class Perf_BitOperations
+    {
+        private static Random s_random = new Random(5);
+        private static uint[] input_uint = GenerateRandomValues<uint>();
+        private static ulong[] input_ulong = GenerateRandomValues<ulong>();
+
+        private static T[] GenerateRandomValues<T>() where T : struct
+        {
+            T[] randomValues = new T[1000];
+            for (int i = 0; i < 1000; i++)
+            {
+                var randomRange = s_random.Next(minValue: 2000 * i, maxValue: 5000 * i);
+                randomValues[i] = Unsafe.As<int, T>(ref randomRange);
+            }
+            return randomValues;
+        }
+
+        [Benchmark]
+        public int LeadingZeroCount_uint()
+        {
+            int sum = 0;
+            for (int i = 0; i < input_uint.Length; i++)
+            {
+                sum += BitOperations.LeadingZeroCount(input_uint[i]);
```

Comment:
nit: JIT is typically producing a better codegen when using local variables rather than fields, it's then 100% sure that nobody can change the value of a field in the meantime.

```suggestion
            int sum = 0;
            uint[] local = input_uint;
            for (int i = 0; i < local.Length; i++)
            {
                sum += BitOperations.LeadingZeroCount(local[i]);
```